lib/auth-middleware/auth_middleware/__init__.py

```diff
@@ -245,23 +346,30 @@ def verify_userauth(id_token, group_name=None, internal_token_only=False):
             logger.warning("invalid issuer = %s" % issuer)
             raise OAuthProblem('Invalid token')
 
-        matching_jwks = [key_dict for key_dict in keyset['keys'] if key_dict['kid'] == kid]
+        matching_jwks = [key_dict for key_dict in keyset['keys']
+                         if key_dict['kid'] == kid]
         if len(matching_jwks) != 1:
             logger.warning("should have exactly one match for kid = %s" % kid)
             raise OAuthProblem('Invalid token')
         jwk = matching_jwks[0]
         pub_key = jwt.algorithms.RSAAlgorithm.from_jwk(json.dumps(jwk))
         try:
-            id_info = jwt.decode(id_token, key=pub_key, audience=target_client_ids, verify=True)
+            if checkAud:
```

Comment:
This never gets set to false. Why is it here?

Reply:
It used to be set to False previously when we used to load pubkeys from the environment, since we didn't have an audience for that.